baseband/dada/payload.py

```diff
@@ -93,3 +97,20 @@ def fromfile(cls, fh, header=None, memmap=False, payloadsize=None,
                               else (payloadsize // cls._dtype_word.itemsize,))
             fh.seek(offset + words.size * words.dtype.itemsize)
         return cls(words, header=header, **kwargs)
+
+    @classmethod
+    def fromdata(cls, data, bps=2):
+        """Encode data as payload.
+
+        Parameters
+        ----------
+        data : `numpy.ndarray`
+            Data to be encoded. The last dimension is taken as the number of
+            channels.
+        bps : int
+            Number of bits per sample to use (for complex data, for real and
+            imaginary part separately; default: 2).
+        """
+        payload = super(DADAPayload, cls).fromdata(data, bps=bps)
+        payload.sample_shape = cls._sample_shape_cls(*data.shape[1:])
```

Comment:
Move to `vlbi_base` and define `_sample_shape_cls = tuple` in there so that it works even if people do not override `_sample_shape_cls`

Reply:
Slight complication - initialization of a `namedtuple` instance is different than that for `tuple`.  I wrote in a try/except block in `VLBIPayloadBase.fromdata` that supports using `namedtuple`, `tuple`, `list` and `np.array` (if a user ever decides to subtype using those as containers).